service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/SdsFilter.java

```diff
@@ -0,0 +1,162 @@
+import org.springframework.http.server.RequestPath;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.web.server.ServerWebExchange;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+import uk.nhs.adaptors.gpc.consumer.sds.SdsClient;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SdsFilter implements GlobalFilter, Ordered {
+    private static final String GPC_URL_ENVIRONMENT_VARIABLE = "GP2GP_GPC_GET_URL";
+    private static final String INTERACTION_ID_PREFIX = "urn:nhs:names:services:gpconnect:";
+    private static final String STRUCTURED_ID = INTERACTION_ID_PREFIX + "fhir:operation:gpc.getstructuredrecord-1";
+    private static final String PATIENT_SEARCH_ID = INTERACTION_ID_PREFIX + "documents:fhir:rest:search:patient-1";
+    private static final String DOCUMENT_SEARCH_ID = INTERACTION_ID_PREFIX + "documents:fhir:rest:search:documentreference-1";
+    private static final String BINARY_READ_ID = INTERACTION_ID_PREFIX + "documents:fhir:rest:read:binary-1";
+    private static final String SSP_INTERACTION_ID = "Ssp-InteractionID";
+    private static final String SLASH = "/";
+    private static final String FHIR_SEPARATOR = "fhir/";
+    private static final String DOCUMENTS_SEPARATOR = "documents/";
+    private static final String DOUBLE_SEPARATOR = DOCUMENTS_SEPARATOR + FHIR_SEPARATOR;
+
+    private final SdsClient sdsClient;
+
+    private Map<String, Function<String, Optional<SdsClient.SdsResponseData>>> sdsRequestFunctions;
+
+    @Override
+    public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+        if (StringUtils.isBlank(System.getProperty(GPC_URL_ENVIRONMENT_VARIABLE))) {
+            ServerHttpRequest serverHttpRequest = exchange.getRequest();
+            extractInteractionId(serverHttpRequest.getHeaders())
+                .ifPresent(id -> proceedSdsLookup(serverHttpRequest, exchange, id));
+        }
+
+        return chain.filter(exchange);
+    }
+
+    @Override
+    public int getOrder() {
+        return RouteToRequestUrlFilter.ROUTE_TO_URL_FILTER_ORDER + 1;
+    }
+
+    @PostConstruct
+    @SuppressWarnings("unused")
+    public void initializeSdsRequestFunctions() {
+        sdsRequestFunctions = Map.of(
+            STRUCTURED_ID, sdsClient::callForGetStructuredRecord,
+            PATIENT_SEARCH_ID, sdsClient::callForPatientSearchAccessDocument,
+            DOCUMENT_SEARCH_ID, sdsClient::callForSearchForDocumentRecord,
+            BINARY_READ_ID, sdsClient::callForRetrieveDocumentRecord);
+    }
+
+    private void proceedSdsLookup(ServerHttpRequest serverHttpRequest,
+            ServerWebExchange exchange,
+            String integrationId) {
+        Optional<SdsClient.SdsResponseData> response
```

Comment:
Don't need optional / isPresent. Just throw an exception as soon as we detect the unsupported interaction id: "Unsupported interaction id: {interactionId}". It is an unrecoverable error.